tests/unit/utils/monitoring.js

```diff
@@ -0,0 +1,69 @@
+const assert = require('assert');
+const promclient = require('prom-client');
+const sinon = require('sinon');
+
+const monitoring = require('../../../lib/utilities/monitoringHandler');
+
+describe('Monitoring: endpoint', () => {
+    const sandbox = sinon.createSandbox();
+    const res = {
+        writeHead(/* result, headers */) { return this; },
+        write(/* body */) { return this; },
+        end(/* body */) {},
+    };
+    monitoring.collectDefaultMetrics();
+
+    beforeEach(() => {
+        sandbox.spy(res);
+        sandbox.spy(promclient.register, 'metrics');
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    async function fetchMetrics(req, res) {
+        await new Promise(resolve => monitoring.monitoringHandler(null, req, {
+            ...res, end: (...body) => { res.end(...body); resolve(); }
+        }, null));
+    }
+
+    it('it should return an error is method is not GET', async () => {
+        await fetchMetrics({ method: 'PUT', url: '/metrics' }, res);
+        assert(res.writeHead.calledOnceWith(405));
+        assert(res.end.calledOnce);
+    });
+
+    it('it should return an error is path is not /metrics', async () => {
+        await fetchMetrics({ method: 'GET', url: '/foo' }, res);
+        assert(res.writeHead.calledOnceWith(405));
+        assert(res.end.calledOnce);
+    });
+
+    it('it should return some metrics', async () => {
+        await fetchMetrics({ method: 'GET', url: '/metrics' }, res);
+        assert(res.writeHead.calledOnceWith(200) || !res.writeHead.called);
```

Comment:
using `||` for assertions feels weird, we should know which one it is

Reply:
i agree, though i did this to match the behavior of http module... (and thus not depend on the actual implementation).
Bit  fixed the code to explicitely call `writeHead`, so no reason to keep it indeed.